### DISCUSSION TITLE
"TypeError : Argument entity is required" shows  when responding with an array element

### DISCUSSION THREAD

**Discussion by jbryan11** [vercel/next.js#19017] [Help]
This wasted my time figuring out what's the problem in my API. My code was like this:

```
import {traffic} from '../../../lib/trafficData'
export default function dataTrafficHandler({query: {dataId}},res){
     if (dataId){
      var filtered = traffic.filter(data=> data.dataId == dataId)
      
       return res.status(200).json(filtered[0]) 
}}
```

I'm trying return a response with an object(which is an array element) from the filtered array. When I dynamically route to the page that uses this API, it doesn't have a problem, but when I reload the page or dynamically enter the URL of the page that uses this api, it shows the `"TypError : Argument entity is required"`

What's intruiging for me is that  when I use spread operator on the `filtered[0]` array element and store it in a a variable and pass it into the response call, it works

```
import {traffic} from '../../../lib/trafficData'
export default function dataTrafficHandler({query: {dataId}},res){
     if (dataId){
      var filtered = traffic.filter(data=> data.dataId == dataId)
       var data = {...filtered[0]}
       return res.status(200).json(data)
     }
}
```

I don't know if its a false report or its really a problem in the entity on my fetch call,

**Comment by anges244** [comment]
This error occurs when the response body is empty. Not sure about your app's specifics but for some reason the data in your res.status(200).json(data) is empty when you attempt to send it. That's why it works when you have it stored. 

Look into what you are trying to send when you reload the page, even with a simple console.log() and verify it's empty.
